src/cpp/core/file_lock/FileLock.cpp

```diff
@@ -209,7 +211,7 @@ void FileLock::log(const std::string& message)
 }
 
 // default values for static members
-FileLock::LockType FileLock::s_defaultType(FileLock::LOCKTYPE_LINKBASED);
+FileLock::LockType FileLock::s_defaultType(kLockTypeDefaultEnum);
```

Comment:
If I'm reading this correctly, it not only makes link based locks possible, it also switches to link based locks by default on MacOS and Linux desktops. Is that intentional? (IIRC we don't have known issues with advisory locks there)

Reply:
That's a good point -- we don't want to change the default here; just make it possible to respect the user-set lock for session locks. I'll fix this.